### PR TITLE
crypto/message-digest: add common infrastructure, improve non-threaded.

### PR DESCRIPTION
**Pull request by barbieri**
this moves the common bits from linux-kcapi and openssl to a single
file, allows that cumbersome core to be shared with other
implementations.

it also introduces a maximum limit per feed if not using threads, this
way we avoid locking the main loop for too long.

Signed-off-by: Gustavo Sverzut Barbieri gustavo.barbieri@intel.com

### CODE REVIEW

**Comment by barbieri** [comment]
updated to include tests and a fix to coverity issue CID 131380

File: src/test/test-message-digest.c [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+
+static void
+test_md5_single(void)
+{
+    /* note: mem includes string trailing null byte */
+    static const struct md_test md5_single_test[] = {
+        MD_TEST("md5", SOL_STR_SLICE_EMPTY, "test", "e2a3e68d23ce348b8f68b3079de3d4c9"),
+        MD_TEST("md5", SOL_STR_SLICE_EMPTY, "long line of text bla bla bla more text here yada yada", "18511ce4f220de4744390ca3ae72873f"),
+        MD_TEST("md5", SOL_STR_SLICE_EMPTY, big_blob_of_zeros, "ab893875d697a3145af5eed5309bee26"),
+        MD_TEST("md5", SOL_STR_SLICE_EMPTY, big_blob_of_chars, "9a36eacb09f8e98e103e9ee897f8e31c"),
```
Comment (lpereira) [inline src/test/test-message-digest.c]: I"d also test for the empty string (which should evaluate to `d41d8cd98f00b204e9800998ecf8427e`).

**Comment by ibriano** [comment]
Seems fine

**Comment by barbieri** [comment]
pushed to master a rebased and reordered patch (moving the kcapi fix earlier).

It adds the hash for zero-sized memory as requested by @lpereira.

It fixes a bug for the chunked content test that would leave a timer to be called after the context was deleted, causing a bug in the test code (not the tested code). Now keep a reference to the timer and delete it alongside with chunked_ctx.
